Manage Logon Hour via Windows PowerShell

```diff
@@ -0,0 +1,345 @@
+
+Changing any value from 1 to 0 or 0 to 1 will be reflected in the Logon Hours GUI. Let's try to change the first bit in the first block, make it 0, and then see the result. 
+
+So change the first block and make it **01111111**, click on OK to save, and open back the Logon Hour in the **Account** tab. See that there is only one block that is now marked as white.
+
+![Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%203.png](Manage%20Logon%20Hour%20via%20Windows%20PowerShell%20c28909d216f44708aad42cfa6706b403/Untitled%203.png)
+
+Result After Changing 1 bit
+
+In this post, what you see might "*till this point*" not look like what you see on your side, even after doing the same change, because the **LogonHour** property uses UTC standard time format.
+
+I will explain more about this issue and how to fix it via the PowerShell script.
+
+## Parameter for the PowerShell Script.
+
+To make things easy to use for the user, I wrote a function **Set-LogonHours** that accepts the following parameter.
+
+- **Identity:** The Username of the target principal (*String, Accept Pipeline, Position=0)*
+- **TimeIn24Format:** Array of the hours access is permitted (*Array, ValidateRange (0- 23)*
+- **Sunday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Monday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Tuesday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Wednesday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Thursday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Friday:** Apply the TimeIn24Format to this day *(Switch)*
+- **Saturday:** Apply the TimeIn24Format to this day *(Switch)*
+- **NonSelectedDaysAre:** This option for the weekdays that are not selected, what the default value should be, is it a permit or denied *(ValidateSet ="WorkingDays" or "NonWorkingDays").*
+
+    ```powershell-console
+    Set-LogonHours -identity "MyTestUser" -TimeIn24Format @(8,9,10,11,12)  -Monday -Tuesday -Wednesday -Thursday -NonSelectedDaysare WorkingDays
+    ```
+
+## Put it All in Action
+
+First, create a Byte array, which will hold the final result as the **LogonHour** accepts a Byte array, not an integer array. The array length is 21, which represents a week. The number of Seven * (three blocks "which represent a day")  (7x3=21)
+
+```powershell-console
+$FullByte=New-Object "byte[]" 21
+```
+
+Then create a Hashtable that represents each hour of the day. The value of this hashtable will be set to **One** based on the user input
+
+```powershell-console
+$FullDay=[ordered]@{
+        "1"=0
+        "2"=0
+        "3"=0
+        "4"=0
+        "5"=0
+        "6"=0
+        "7"=0
+        "8"=0
+        "9"=0
+        "10"=0
+        "11"=0
+        "12"=0
+        "13"=0
+        "14"=0
+        "15"=0
+        "16"=0
+        "17"=0
+        "18"=0
+        "19"=0
+        "20"=0
+        "21"=0
+        "22"=0
+        "23"=0
+        "0"=0
+        }
```

Comment:
AWFUL code.
At a minimum:
```powerShell
$FullDay=[ordered]@{}
1..24 | Foreach-Object {$FullDay[$_]=0}

Reply:
Hi, 
The original code was actually 
`$FullDay=[ordered]@{}
0..23 | foreach{$FullDay.Add($_,"0")}`
But l make it like this for the user to understand this hashtable and its representation. 
Anyway, I fix it back